test-utils/src/dev/clickhouse.rs

```diff
@@ -147,45 +333,66 @@ impl ClickHouseInstance {
             "--config-file".to_string(),
             format!("{}", config_path.display()),
         ];
+        let mut env: BTreeMap<_, _> = [
+            (String::from("CLICKHOUSE_WATCHDOG_ENABLE"), String::from("0")),
+            (String::from("CH_LOG"), data_dir.log_path().to_string()),
+            (String::from("CH_ERROR_LOG"), data_dir.err_log_path().to_string()),
+            (String::from("CH_REPLICA_DISPLAY_NAME"), name.clone()),
+            (String::from("CH_LISTEN_ADDR"), String::from("::")),
+            (String::from("CH_LISTEN_PORT"), port.to_string()),
+            (String::from("CH_TCP_PORT"), tcp_port.to_string()),
+            (String::from("CH_INTERSERVER_PORT"), interserver_port.to_string()),
+            (
+                String::from("CH_DATASTORE"),
+                data_dir.datastore_path().to_string(),
+            ),
+            (String::from("CH_TMP_PATH"), data_dir.tmp_path().to_string()),
+            (
+                String::from("CH_USER_FILES_PATH"),
+                data_dir.user_files_path().to_string(),
+            ),
+            (
+                String::from("CH_USER_LOCAL_DIR"),
+                data_dir.access_path().to_string(),
+            ),
+            (
+                String::from("CH_FORMAT_SCHEMA_PATH"),
+                data_dir.format_schemas_path().to_string(),
+            ),
+            (String::from("CH_REPLICA_NUMBER"), r_number.clone()),
+            (String::from("CH_REPLICA_HOST_01"), String::from("::1")),
+            (String::from("CH_REPLICA_HOST_02"), String::from("::1")),
+            // ClickHouse servers have a small quirk, where when setting the
+            // keeper hosts as IPv6 localhost addresses in the replica
+            // configuration file, they must be wrapped in square brackets
+            // Otherwise, when running any query, a "Service not found" error
+            // appears.
+            (String::from("CH_KEEPER_HOST_01"), String::from("[::1]")),
+            (String::from("CH_KEEPER_HOST_02"), String::from("[::1]")),
+            (String::from("CH_KEEPER_HOST_03"), String::from("[::1]")),
+        ]
+        .into_iter()
+        .collect();
```

Comment:
I think in the future we can use clickward to generate the configuration files for these tests, it's a lot cleaner than setting a bunch of environment variables. But that's unrelated to this PR, no need to change anything now!